controllers/openstackdataplanedeployment_controller.go

```diff
@@ -221,6 +222,15 @@ func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context,
 			AeeSpec:         &ansibleEESpec,
 			InventorySecret: nodeSetSecretInv,
 		}
+		// Merge Ansible extra variables from the referenced NodeSet with
+		// the Ansible variables from the OpenStackDataPlaneDeployment
+		if instance.Spec.AnsibleExtraVars != nil {
+			for k, v := range instance.Spec.AnsibleExtraVars {
+				if _, ok := ansibleEESpec.ExtraVars[k]; !ok {
+					ansibleEESpec.ExtraVars[k] = v
+				}
+			}
+		}
```

Comment:
You've already copied from instance.Spec.AnsibleExtraVars to ansibleEESpec.ExtraVars at line 205. I think you can do away with this ```if``` block as there's no need to merge variables from the NodeSet.